odoo/tools/profiler.py

```diff
@@ -166,20 +171,34 @@ def __init__(self, interval=0.01):  # check duration. dynamic?
 
     def run(self):
         self.active = True
+        last = time.time()
         while self.active:  # maybe add a check on parent_thread state?
+            frame_time = time.time() - last
+            if frame_time > self.frame_interval*10 and self.last_frame:
```

Comment:
```suggestion
            if frame_time > self.frame_interval * 10 and self.last_frame:
                # The profiler has unexpectedly slept for more than 10 frame intervals. This may
                # happen when calling a C library without releasing the GIL. In that case, the
                # last frame was taken before the call, and the next frame is after the call, and
                # the call itself does not appear in any of those frames: the duration of the call
                # is incorrectly attributed to the last frame.
```